### DISCUSSION TITLE
Assign error codes for various GDScript compile errors

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#4122] [Scripting]
In addition to text error messages, add error codes to them. For example `LNK2001: unresolved external symbol`.

This will make life easier if we finally localize error messages.

**Comment by Calinou** [comment]
This is a good idea to improve searchability of error messages in general. One open question that remains is, how to *name* those error codes:

1. Use human-readable English names like `unresolved_external_symbol`. This immediately gives you an idea of what the error code is about. This appears to be what most linters like ESLint do. GDScript warnings already have those snake_case identifiers for the `@warning_ignore()` annotation.
2. Alternatively, use error codes like `GDS1001`. This wouldn't match what GDScript warnings currently do, though.

**Comment by Mickeon** [comment]
GDScript aims to be very user-friendly, so arbitrary numerical error codes over a human-readable ones may be "alienating", unless they're internally associated to the error specifically for localization purposes.

**Reply by timothyqiu** [reply]
This is not about replacing `Unexpected indentation.` with `GDS1001`, but replacing it with `GDS1001: Unexpected indentation.`.

So we can give localized error messages like `GDS1001: Indentation inattendue.` and `GDS1001: 意外的缩进。`, and still allow these users to search for that error in a wider community.
